_search-plugins/knn/api.md

```diff
@@ -175,6 +175,48 @@ For the warmup operation to function properly, follow these best practices:
 
 * Don't index any documents that you want to load into the cache. Writing new information to segments prevents the warmup API operation from loading the native library indexes until they're searchable. This means that you would have to run the warmup operation again after indexing finishes.
 
+## k-NN Clear Cache
+Introduced 2.14
+{: .label .label-purple }
+
+During approximate k-Nearest Neighbor (k-NN) search or warmup operation, the native library indices(for `nmslib` and `faiss` engines) are loaded into native memory. Currently, the possible ways to evict an index from cache or native memory is by either deleting the index or by setting a 
+couple of k-NN cluster settings(`knn.cache.item.expiry.enabled` and `knn.cache.item.expiry.minutes`) which will remove the index out of the cache if they are idle for a given amount of time. But, there is no way to evict a given index out of cache without deleting the index. To solve this problem, 
+we can use the k-NN clear cache api operation which helps to clear given set of indices out of the cache.
+
+This clear cache api evicts all the native library files for all the shards(primaries and replicas) of all the indices specified in the request. Like [warmup operation](#warmup-operation) this k-NN clear cache api is idempotent where if an index has been already evicted from cache it has no impact.
+
+### Usage
+This request evicts the native library indices of these three indices from cache:
+```json
+GET /_plugins/_knn/clear_cache/index1,index2,index3?pretty
```

Comment:
Do we add query parameters anywhere?

Reply:
Sorry didn't get it. Are you referring to [REST parameters](https://opensearch.org/docs/2.3/opensearch/common-parameters/) ?

Reply:
Yes, I would say we need to call out what index1,... are and also say that they are optional. If not passed, all memory will be cleared.

Reply:
no, it is not optional. Users needs to pass the indices as part of uri. To remove all the indices from cache we have another path `/_plugins/_knn/clear_cache/_all` but the problem with this is it fetches all the indices in the domain and validates if each index is of type knn. If any index is not of type knn then it will throw exception and reject the request. It might not work most of the cases so we are not including it in the doc and thought that we can recommend it if any Cx reaches out to us asking for such requirement.